spec/connectors/mongodb/connector_spec.rb

```diff
@@ -292,5 +432,203 @@
         end
       end
     end
+
+    shared_examples_for 'logs a warning on yield documents' do
+      it 'logs a warning' do
```

Comment:
why?

To make this clear to readers in the future, maybe augment this with an expectation of what the warning log message should be.